shell.nix

```diff
@@ -1,34 +1,50 @@
-{ stdenv, pkgs }:
+{ pkgs ? import <nixpkgs> { } }:
+with pkgs;
+let
+  go = go_1_19;
+  postgresql = postgresql_14;
+  nodejs = nodejs-16_x;
+in
+mkShell {
+  nativeBuildInputs = [
+    go
 
-pkgs.mkShell {
-  nativeBuildInputs = with pkgs; [
-    go_1_17
-
-    postgresql_13
+    postgresql
     python3
     python3Packages.pip
     curl
-    nodejs-16_x
-    (yarn.override { nodejs = nodejs-16_x; })
+    nodejs
+    (yarn.override { inherit nodejs; })
     # TODO: compiler / gcc for secp compilation
-    nodePackages.ganache-cli
+    nodePackages.ganache
     # py3: web3 slither-analyzer crytic-compile
     # echidna
-    # go-ethereum # geth
+    go-ethereum # geth
     # parity # openethereum
-    # go-mockery
+    go-mockery
 
     # tooling
-    goimports
+    gotools
     gopls
     delve
     golangci-lint
+    github-cli
 
     # gofuzz
+  ] ++ lib.optionals stdenv.isLinux [
+    # some dependencies needed for node-gyp on yarn install
+    pkg-config
+    libudev-zero
+    libusb1
   ];
-  LD_LIBRARY_PATH="${stdenv.cc.cc.lib}/lib64:$LD_LIBRARY_PATH";
-  GOROOT="${pkgs.go_1_17}/share/go";
+  LD_LIBRARY_PATH = "${stdenv.cc.cc.lib}/lib64:$LD_LIBRARY_PATH";
+  GOROOT = "${go}/share/go";
 
-  PGDATA="db";
+  PGDATA = "db";
+  DATABASE_URL = "postgresql://chainlink:chainlink@localhost:5432/chainlink_test?sslmode=disable";
+  shellHook = ''
+    export GOPATH=$HOME/go
```

Comment:
Hmm, is this required? Could maybe use `export GOPATH=$(${go}/bin/go env GOPATH)`

Reply:
Actually, both `GOROOT` and this `shellHook` should be removed, they're not needed anymore